assets/js/modules/adsense/datastore/adblocker.js

```diff
@@ -134,6 +140,46 @@ export const selectors = {
 		const { isAdBlockerActive } = state;
 		return isAdBlockerActive;
 	},
+
+	/**
+	 * Returns appropriate ad blocker warning message based on modules connection status.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|null} The error message or null if ad blocker isn't detected.
+	 */
+	getAdBlockerWarningMessage: Data.createRegistrySelector(
+		( select ) => () => {
+			const isAdBlockerActive = select(
+				MODULES_ADSENSE
+			).isAdBlockerActive();
+
+			if ( undefined === isAdBlockerActive ) {
+				return undefined;
+			}
+
+			if ( ! isAdBlockerActive ) {
+				return null;
+			}
+
+			const isModuleConnected = select( CORE_MODULES ).isModuleConnected(
+				'adsense'
+			);
+
+			if ( isModuleConnected ) {
+				return __(
+					'Ad blocker detected, you need to disable it to get the AdSense latest data.',
```

Comment:
I know this string was originally written this way, but it's awkward. Let's take this opportunity to improve it, because "AdSense latest data" is not very clearly written:

```suggestion
					'Ad blocker detected, you need to disable it to get the latest AdSense data.',
```